src/System.Threading.ThreadPool/tests/ThreadPoolTests.cs

```diff
@@ -14,6 +14,43 @@ public static class ThreadPoolTests
         private const int ExpectedTimeoutMilliseconds = ThreadTestHelpers.ExpectedTimeoutMilliseconds;
         private const int MaxPossibleThreadCount = 0x7fff;
 
+        static ThreadPoolTests()
+        {
+            // Run the following tests before any others
+            ConcurrentInitializeTest();
+        }
+
+        [Fact]
+        public static void ConcurrentInitializeTest()
+        {
+            int processorCount = Environment.ProcessorCount;
+            var threadReady = new AutoResetEvent(false);
+            var continueThreads = new ManualResetEvent(false);
+            Action threadMain =
+                () =>
+                {
+                    threadReady.Set();
+                    continueThreads.CheckedWait();
+                    Assert.True(ThreadPool.SetMinThreads(processorCount, processorCount));
+                };
+
+            var waitForThreadArray = new Action[processorCount];
+            for (int i = 0; i < processorCount; ++i)
+            {
+                var t = ThreadTestHelpers.CreateGuardedThread(out waitForThreadArray[i], threadMain);
+                t.IsBackground = true;
+                t.Start();
+                threadReady.CheckedWait();
+            }
+
+            continueThreads.Set();
+
+            foreach (Action waitForThread in waitForThreadArray)
+            {
+                waitForThread();
```

Comment:
Doing synchronization in a static cctor can easily lead to deadlock.  May not be an issue here, but something to keep in mind.